### ISSUE TITLE
timingSafeEqual problems with a GET endpoint

### ISSUE THREAD

**Issue by jiferrero** [connorjburton/hmac-auth-express#85]
Morning Mr. Connor,

Thanks for your wonderful package! We are currently using it with no problems in POST routes until we try to use a GET route.
Verification always fails with _"HMAC's did not match"_ because it not pass the **timingSafeEqual** test. We are struggling our heads why did not do because if we use the same endpoint but with a POST verb it does.
Can you give us a hand with this? A small ray of light on this will be appreciated.
Thank you so much

**Comment by connorjburton** [comment]
Hi @jiferrero - this is most likely because you aren't setting a body on when generating your HMAC. The way this package works is that it adds the contents of whatever `req.body` is, [see the logic here](https://github.com/connorjburton/hmac-auth-express/blob/master/src/generate.ts#L42).

The exact behaviour of this package is dependent on your configuration in your express application. from what I remember, if you are using the external [`body-parser`](https://www.npmjs.com/package/body-parser) package then a GET `req.body` will be `undefined`, if you are using the built-in body parser it will be `{}` as [per the documentation](http://expressjs.com/en/api.html#express.json).

> A new body object containing the parsed data is populated on the request object after the middleware (i.e. req.body), or an empty object ({}) if there was no body to parse, the Content-Type was not matched, or an error occurred.

[See this previously raised issue](https://github.com/connorjburton/hmac-auth-express/issues/82)

To sum up, I first would try logging out in `node_modules/hmac-auth-express/dist/generate.js` what exactly `request.body` is and then match your HMAC generation to be that, you probably need to set the body to `{}` (or not use the body parser on GET requests, which is more sensible)

Hope that helps! If it doesn't let me know.

Thanks

**Comment by jiferrero** [comment]
Thank you so much @connorjburton
We are using express with external body-parser so we are going to do some adjustments and tests in the way you suggest it.

Cross fingers and let you know about!

**Comment by jiferrero** [comment]
Hi @connorjburton .... all solved right now.... our mistake.
We were doing it right from first time because we have analyzed and track your code... but we forgot to update the endpoint VERB when making the endpoint call so we were still using POST when calling a GET. 
We feel a little stupid right now! :)

Thank you so much anyway for your quickly respond and wish you a **Happy New Year**!

**Comment by connorjburton** [comment]
@jiferrero No problem, and thank you for using the package! Great to see it's being used :)

I do plan to write some better auto-debugging at some point, as it can be very tricky to figure out exactly what part of the HMAC is wrong, some form of opt-in flag that tells you exactly the diff would be good, I think...

Happy new year to you too!
